src/NuGet.Core/NuGet.Protocol/Utility/LocalFolderUtility.cs

```diff
@@ -987,13 +987,13 @@ private static string EnsurePackageExtension(string packagePath, bool isSnupkg)
         }
 
         /// <summary>
-        /// This method checks of a given list of package paths exist on disk.
+        /// If there isn't at least one Path specified, throw that no file paths were resolved for this Package.
         /// </summary>
-        /// <param name="packagePath">A package path to be used while creating logs.</param>
-        /// <param name="matchingPackagePaths">A list of matching package paths that need to be checked.</param>
+        /// <param name="packagePath">The package path the user originally provided.</param>
+        /// <param name="matchingPackagePaths">A list of matching package paths that were previously resolved.</param>
         public static void EnsurePackageFileExists(string packagePath, IEnumerable<string> matchingPackagePaths)
         {
-            if (!matchingPackagePaths.Any())
+            if (!(matchingPackagePaths != null && matchingPackagePaths.Any()))
```

Comment:
```suggestion
            if (matchingPackagePaths == null || !matchingPackagePaths.Any())
```

Personally I find this easier to understand (no double negative)

Reply:
I agree the other way is more pleasing to read, but I'm going to maintain the bundled null-check inside the nested parens. Harder to introduce a tricky bug when someone eventually modifies one of these if-statements.